### PR TITLE
third-party: upgrade libluv: 1.30.1-1 => 1.34.2-2

### PR DESCRIPTION
**Pull request by 0x777**
I'm not very confident about the Makefile changes (required because of some upstream changes).

### CODE REVIEW

File: third-party/cmake/BuildLuarocks.cmake [hunk 1]
```diff
@@ -203,7 +203,7 @@ if(USE_BUNDLED_BUSTED)
     if(USE_BUNDLED_LIBUV)
       list(APPEND LUV_ARGS LIBUV_DIR=${HOSTDEPS_INSTALL_DIR})
     endif()
-    SET(LUV_PRIVATE_ARGS LUA_COMPAT53_INCDIR=${DEPS_BUILD_DIR}/src/lua-compat-5.3)
+    SET(LUV_PRIVATE_ARGS LUA_COMPAT53_INCDIR=${DEPS_BUILD_DIR}/src/lua-compat-5.3/c-api)
```
Comment (0x777) [inline third-party/cmake/BuildLuarocks.cmake:206]: https://github.com/luvit/luv/commit/dec46617cd4781bbdf4c0f9fa0d2d3e01779e930

File: cmake/FindLibLUV.cmake [hunk 2]
```diff
@@ -15,7 +15,7 @@ find_path(LIBLUV_INCLUDE_DIR luv/luv.h
           PATHS ${PC_LIBLUV_INCLUDEDIR} ${PC_LIBLUV_INCLUDE_DIRS})
 
 # Explicitly look for luv.so. #10407
-list(APPEND LIBLUV_NAMES luv luv${CMAKE_SHARED_LIBRARY_SUFFIX})
+list(APPEND LIBLUV_NAMES luv luv_a luv${CMAKE_SHARED_LIBRARY_SUFFIX})
```
Comment (0x777) [inline cmake/FindLibLUV.cmake:18]: https://github.com/luvit/luv/pull/461

File: third-party/cmake/BuildLuv.cmake [hunk 3]
```diff
@@ -71,6 +71,7 @@ set(LUV_CONFIGURE_COMMAND_COMMON
   -DLUA_COMPAT53_DIR=${DEPS_BUILD_DIR}/src/lua-compat-5.3
   -DWITH_SHARED_LIBUV=ON
   -DBUILD_SHARED_LIBS=OFF
+  -DBUILD_STATIC_LIBS=ON
```
Comment (0x777) [inline third-party/cmake/BuildLuv.cmake:74]: https://github.com/luvit/luv/pull/461

**Comment by clason** [comment]
If you are still interested in carrying this forward, I'd suggest considering bumping this even further:
* libluv 1.36.0 makes some interesting async fixes and allows building against older libuv versions (which might be useful).
* libluv 1.40.0 has a number of memory leaks fixed.

But 1.34.2-2 is the big, breaking change, and it makes sense to start with that.

If you rebase on master, I'll run the checks again.

**Comment by clason** [comment]
I tested locally on macOS, and these changes build both 1.34.2-2 and 1.40.0-0 successfully -- take your pick. If you prefer not to continue this PR, let me know and I'll open a new PR myself.

**Comment by clason** [comment]
Sorry for not waiting; I got curious myself and had to open my own PR to test things. The key change needed was to [add `libluv_a` as static link target](https://github.com/luvit/luv/releases/tag/1.34.2-1) (not sure why that wasn't needed on macOS or Linux, and _only_ specifying `luv_a` in fact is an error 🤷🏻‍♂️ ).

I hope you don't mind; otherwise I can close my PR and let you add this change here.

**Comment by 0x777** [comment]
> I hope you don't mind; otherwise I can close my PR and let you add this change here.

Nope, not at all :) Thanks for taking a looking at this. I'm closing this in favor of https://github.com/neovim/neovim/pull/14997.

**Pull request closed without merging**